.github/workflows/ci.yml

```diff
@@ -101,46 +93,15 @@ jobs:
       - name: Install dependencies
         uses: ramsey/composer-install@v2
 
-      - name: Pull PHPStan cache
-        id: pull-phpstan-cache
-        uses: actions/cache/restore@v3
+      - name: Cache PHPStan
+        uses: actions/cache@v3
         with:
           path: ${{ env.PHPSTAN_CACHE_DIR }}
-          key: phpstan-${{ github.ref_name }}
-          restore-keys: |
-            phpstan
+          key: phpstan-${{ runner.os }}-${{ steps.setup_php.outputs.php-version }}
 
       - name: Run PHPStan
         run: composer run stan
 
-      - name: Install gh-actions cache plugin
-        if: always()
-        run: gh extension install actions/gh-actions-cache
-
-      - name: Clear previous version of PHPStan branch cache
-        if: always() # We always delete the cache, because even if it was not 'hit', it might have been created by a commit in the meantime
-        continue-on-error: true
-        run: gh actions-cache delete phpstan-${{ github.ref_name }} -B ${{ github.ref }} --confirm
-
-      - name: Push PHPStan branch cache
-        if: always()
-        uses: actions/cache/save@v3
-        with:
-          path: ${{ env.PHPSTAN_CACHE_DIR }}
-          key: phpstan-${{ github.ref_name }}
-
-      - name: Clear previous version of PHPStan main cache
```

Comment:
The reason we clear the previous version of the cache is because Github actions caches live for 90 days. So if we have a cache, it won't get updated for 90 days, after which it will get deleted and not be present for the next run, after which it will be up to date again, repeating the sequence.

Reply:
> So if we have a cache, it won't get updated for 90 days

No :)
The cache is updated on every successful run.

![image](https://github.com/PrinsFrank/standards/assets/952007/8b2cb6f7-e74a-4894-8547-c46d0cfc157d)

Reply:
> > So if we have a cache, it won't get updated for 90 days
> 
> No :) The cache is updated on every successful run.

I've just rerun the pipeline for this branch, and now I get this:

![afbeelding](https://github.com/PrinsFrank/standards/assets/25006490/dc40ac92-a464-420a-8481-80d68c1d69bd)

So for every first run for a branch a cache will be created, afterwards the cache will not be updated until it expires and the cache key does not exist anymore.

Reply:
Can be easily checked, you updated some files in recent commits and the cache is not updated:
![afbeelding](https://github.com/PrinsFrank/standards/assets/25006490/cb2b07c2-457e-4677-906d-857086d72cde)

Github actions caches are immutable: https://github.com/actions/cache/blob/main/tips-and-workarounds.md#update-a-cache

Reply:
# 🪫

I am low on batteries.

This is all I can offer today.